pkg/k8s/scaledobject.go

```diff
@@ -22,21 +24,27 @@ func NewScaledObjectClient(cl dynamic.Interface) dynamic.NamespaceableResourceIn
 }
 
 // DeleteScaledObject deletes a scaled object with the given name
-func DeleteScaledObject(name string, cl dynamic.ResourceInterface) error {
-	return cl.Delete(name, &v1.DeleteOptions{})
+func DeleteScaledObject(ctx context.Context, name string, cl dynamic.ResourceInterface) error {
```

Comment:
@khaosdoctor can you update this to use `client.Client` instead of the the dynamic client?

Reply:
Done on 66df110